src/main/java/org/broadinstitute/hellbender/tools/funcotator/Funcotator.java

```diff
@@ -369,22 +435,51 @@ public void closeTool() {
      */
     private void enqueueAndHandleVariant(final VariantContext variant, final ReferenceContext referenceContext, final FeatureContext featureContext) {
 
+        //TODO: Check against Issue #4358.
+        // Check to see if we're annotating filtered variants and ignore this if we're told to:
+        if ( ignoreFilteredVariants && variant.isFiltered() ) {
+            // We can ignore this variant since it was filtered out.
+            return;
+        }
+
         // Get our feature inputs:
         final List<Feature> featureList = new ArrayList<>();
-        for ( final FeatureInput<? extends Feature> featureInput : manualFeatureInputs ) {
-            featureList.addAll( featureContext.getValues(featureInput) );
+
+        // Create a variant context for annotation that has a new contig based on whether we need to overwrite the
+        // contig names in the next section.
+        final VariantContextBuilder variantContextBuilderForFixedContigForDataSources = new VariantContextBuilder(variant);
```

Comment:
This is actually a really good point.  Since the input file is assumed to be sorted, most cases will have the same contig.

That said, I think I'd like to wait on the optimization.  I'll add a `todo` in the code.